src/main/java/focusedCrawler/link/classifier/builder/LinkClassifierBuilder.java

```diff
@@ -108,45 +112,31 @@ public synchronized LinkClassifier forwardlinkTraining(Set<String> relUrls, int
 	}
 
     public LinkClassifier createLinkClassifierImpl(String[] attributes, String[] classValues,
-            Classifier classifier, String className, int levels) {
+    		SoftClassifier<double[]> classifier, String className, int levels) {
 
         LinkNeighborhoodWrapper wrapper = new LinkNeighborhoodWrapper(attributes, stoplist);
 
-        weka.core.FastVector vectorAtt = new weka.core.FastVector();
-        for (int i = 0; i < attributes.length; i++) {
-            vectorAtt.addElement(new weka.core.Attribute(attributes[i]));
-        }
-        weka.core.FastVector classAtt = new weka.core.FastVector();
-        for (int i = 0; i < classValues.length; i++) {
-            classAtt.addElement(classValues[i]);
-        }
-        vectorAtt.addElement(new weka.core.Attribute("class", classAtt));
-        Instances insts = new Instances("link_classification", vectorAtt, 1);
-        insts.setClassIndex(attributes.length);
-
         LinkClassifier linkClassifier = null;
         if (className.indexOf("LinkClassifierImpl") != -1) {
-            LNClassifier lnClassifier = new LNClassifier(classifier, insts, wrapper, attributes);
+        	LNClassifier lnClassifier = new LNClassifier(classifier, wrapper, attributes);
             linkClassifier = new LinkClassifierImpl(lnClassifier);
         }
         if (className.indexOf("LinkClassifierAuthority") != -1) {
-            linkClassifier = new LinkClassifierAuthority(classifier, insts, wrapper, attributes);
+            linkClassifier = new LinkClassifierAuthority(classifier,  wrapper, attributes);
         }
         if (className.indexOf("LinkClassifierHub") != -1) {
-            linkClassifier = new LinkClassifierHub(classifier, insts, wrapper, attributes);
+            linkClassifier = new LinkClassifierHub(classifier, wrapper, attributes);
         }
         return linkClassifier;
     }
     
-    public Classifier trainWekaClassifier(String wekaInputAsString) throws Exception {
-        Instances data = null;
-        try(StringReader reader = new StringReader(wekaInputAsString)) {
-            data = new Instances(reader);
+    public SoftClassifier<double[]> trainSmileClassifier(AttributeDataset smileInput ) throws Exception {
+        SVM<double[]> classifier = new SVM<double[]>(new LinearKernel(), 1.0);
+        int[] y = new int[smileInput.y().length];
+        for(int i = 0 ; i < y.length; i++) {
+        	y[i] = (int) smileInput.y()[i];
         }
-        data.setClassIndex(data.numAttributes() - 1);
-        Classifier classifier = new weka.classifiers.functions.SMO();
-        classifier.setOptions(weka.core.Utils.splitOptions("-C 1.0 -L 0.0010 -P 1.0E-12 -N 0 -M -V -1 -W 1 -K \"weka.classifiers.functions.supportVector.PolyKernel -C 250007 -E 1.0\" -no-cv"));
-        classifier.buildClassifier(data);
+        classifier.learn(smileInput.x(), y);

```

Comment:
Platt's scalling needs to be trained here too, otherwise scores will not be valid!